pkg/asset/machines/vsphere/machines.go

```diff
@@ -22,15 +23,49 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	if poolPlatform := pool.Platform.Name(); poolPlatform != vsphere.Name {
 		return nil, fmt.Errorf("non-VSphere machine-pool: %q", poolPlatform)
 	}
+
+	var machines []machineapi.Machine
+
 	platform := config.Platform.VSphere
 	mpool := pool.Platform.VSphere
 
-	total := int64(1)
+	azs := mpool.Zones
+	numOfAZs := len(azs)
+	definedZones := make(map[string]*vsphere.Platform)
+
+	if numOfAZs > 0 {
+		for _, az := range azs {
+			for _, deploymentZone := range platform.DeploymentZones {
+				if az == deploymentZone.Name {
+					if deploymentZone.ControlPlane == vsphere.NotAllowed {
```

Comment:
Hm I'm not sure about this. Two things:

1. Don't both roles (`compute` & `master`) go through this code path? So I suspect this would incorrectly fail if a zone with `ControlPlane == NotAllowed` were used for a compute machine pool.
2. I suspect this code should live in the machinepool validation code--not here.

Reply:
Machines is only used for rendering control plane nodes.  Compute nodes are rendered from [machinesets](https://github.com/openshift/installer/pull/5911/files/c41f2c10e2929213b597de909dea0fe77a8dd2b6..b14a7a5745406a924280ab9dc8cf0a4e851b5ab3).  I can look at pulling this in to the machinepool validation.

Reply:
its coming back to me now.  at the point of [machinepool validation](https://github.com/openshift/installer/pull/5911/commits/c41f2c10e2929213b597de909dea0fe77a8dd2b6#diff-898a198eb6ce7f9bbd3a05ff17e82ea2e5114d2b7b5694a523f64e99e17f8e5bR11), there isn't a readily available context indicating that a given pool is controlPlane or compute.

Reply:
oh ... wait, we know the pool name, and that is validated to ensure it matches controlPlane and compute pools.

Reply:
@patrickdillon changes have been committed which apply default zones based upon the machine pool role and what deploymentZones are defined.